integrationtests/src/test/java/io/kroxylicious/proxy/ResilienceIT.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    public void producerShouldTolerateSniExposedKroxyliciousRestarting(Admin admin) throws Exception {
+        String randomTopic = UUID.randomUUID().toString();
+        admin.createTopics(List.of(new NewTopic(randomTopic, 1, (short) 1))).all().get();
+        testProducerCanSurviveARestart(sniConfiguration(cluster), randomTopic);
+    }
+
+    @Test
+    public void consumerShouldToleratePortPerBrokerExposedKroxyliciousRestarting(Admin admin) throws Exception {
+        String randomTopic = UUID.randomUUID().toString();
+        admin.createTopics(List.of(new NewTopic(randomTopic, 1, (short) 1))).all().get();
+        testConsumerCanSurviveKroxyliciousRestart(withDefaultFilters(proxy(cluster)), randomTopic);
+    }
+
+    @Test
+    public void consumerShouldTolerateSniExposedKroxyliciousRestarting(Admin admin) throws Exception {
+        String randomTopic = UUID.randomUUID().toString();
+        admin.createTopics(List.of(new NewTopic(randomTopic, 1, (short) 1))).all().get();
+        testConsumerCanSurviveKroxyliciousRestart(sniConfiguration(cluster), randomTopic);
+    }
+
+    private static void testConsumerCanSurviveKroxyliciousRestart(ConfigurationBuilder builder, String topic)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        Producer<String, String> producer;
+        Consumer<String, String> consumer;
+        try (var tester = kroxyliciousTester(builder)) {
+            producer = tester.producer(Map.of(CLIENT_ID_CONFIG, "shouldPassThroughRecordUnchanged", DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000));
+            consumer = tester.consumer();
+            producer.send(new ProducerRecord<>(topic, "my-key", "Hello, world!")).get(10, TimeUnit.SECONDS);
+            consumer.subscribe(Set.of(topic));
+            var records = consumer.poll(Duration.ofSeconds(10));
+            assertEquals(1, records.count());
+            assertEquals("Hello, world!", records.iterator().next().value());
+            producer.send(new ProducerRecord<>(topic, "my-key", "Hello, again!")).get(10, TimeUnit.SECONDS);
+        }
+        try (var ignored = kroxyliciousTester(builder)) {
+            var records = consumer.poll(Duration.ofSeconds(10));
+            assertEquals(1, records.count());
+            assertEquals("Hello, again!", records.iterator().next().value());
+        }
+    }
+
+    private static void testProducerCanSurviveARestart(ConfigurationBuilder builder, String topic) throws InterruptedException, ExecutionException, TimeoutException {
+        Producer<String, String> producer;
+        Consumer<String, String> consumer;
+        try (var tester = kroxyliciousTester(builder)) {
+            producer = tester.producer(Map.of(CLIENT_ID_CONFIG, "shouldPassThroughRecordUnchanged", DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000, RECONNECT_BACKOFF_MS_CONFIG, 5,
+                    RECONNECT_BACKOFF_MAX_MS_CONFIG, 100, RETRY_BACKOFF_MS_CONFIG, 0));
+            consumer = tester.consumer();
```

Comment:
Is there a reason to take the consumer from the first kroxyliciousTester?  I would have expected it to be naturally scoped to the second block.

Reply:
Ok I get it.